src/adapters/RoomsJSONAdapter.js

```diff
@@ -74,9 +76,32 @@ export default class RoomsJSONAdapter extends RoomsAdapter {
     return Observable.create((observer) => {
       const data = !this.datasource[`${ID}-activities`] ? [] : this.datasource[`${ID}-activities`];
 
-      observer.next(data);
+      const formattedData = RoomsJSONAdapter.formatActivity(data);
+
+      observer.next(formattedData);
 
       observer.complete();
     });
   }
 }
+
+/**
+ * Activities could be objects with metadata (like date separators).
+ * Convert the data into proper javascript objects
+ *
+ * @param {Array} activityData json formatted activity data
+ * @returns {Array}
+ */
+RoomsJSONAdapter.formatActivity = (activityData) =>
+  activityData.map((activity) => {
+    let formattedActivity = activity;
+
+    if (typeof activity === 'object') {
```

Comment:
this should be `activityID` I would assume? It was a bit hard for me to realize that this is not actual activity object.

Reply:
updated to be more legible.